### ISSUE TITLE
'MPI_Type_struct is depricated': errors to look forward to

### ISSUE THREAD

**Issue by markyoder** [geodynamics/vq#74]
when compiling vq on kapalua (an osx [not sure of specifics] ) machine, i get the following warnings:
[ 84%] /Users/markyoder/vq/src/core/Comm.cpp:185:5: warning: 'MPI_Type_struct' is deprecated: MPI_Type_struct is superseded by MPI_Type_create_struct in
      MPI-2.0 [-Wdeprecated-declarations]
    MPI_Type_struct(2, block_lengths, displacements, datatypes, &block_val_type);
    ^
/opt/local/include/openmpi-mp/mpi.h:1783:20: note: 'MPI_Type_struct' has been explicitly marked deprecated here
OMPI_DECLSPEC  int MPI_Type_struct(int count, int array_of_blocklengths[],
                   ^
/Users/markyoder/vq/src/core/Comm.cpp:204:5: warning: 'MPI_Type_struct' is deprecated: MPI_Type_struct is superseded by MPI_Type_create_struct in
      MPI-2.0 [-Wdeprecated-declarations]
    MPI_Type_struct(3, block_lengths, displacements, datatypes, &element_sweep_type);
    ^
/opt/local/include/openmpi-mp/mpi.h:1783:20: note: 'MPI_Type_struct' has been explicitly marked deprecated here
OMPI_DECLSPEC  int MPI_Type_struct(int count, int array_of_blocklengths[],

basically, some of the MPI syntax is being deprecated. while these are warnings today, we might keep an eye on them and revise the syntax sooner rather than later, when the opportunity arises, to avoid future code-rot.
